arch/platform/cooja/dev/cooja-radio.c

```diff
@@ -233,24 +223,47 @@ radio_send(const void *payload, unsigned short payload_len)
     return RADIO_TX_ERR;
   }
 
-  /* Transmit on CCA */
-#if COOJA_TRANSMIT_ON_CCA
-  if(send_on_cca && !channel_clear()) {
-    return RADIO_TX_COLLISION;
+  if(radio_was_on) {
+    ENERGEST_SWITCH(ENERGEST_TYPE_LISTEN, ENERGEST_TYPE_TRANSMIT);
+  } else {
+    /* Turn on radio temporarily */
+    simRadioHWOn = 1;
+    ENERGEST_ON(ENERGEST_TYPE_TRANSMIT);
   }
-#endif /* COOJA_TRANSMIT_ON_CCA */
 
-  /* Copy packet data to temporary storage */
-  memcpy(simOutDataBuffer, payload, payload_len);
-  simOutSize = payload_len;
-
-  /* Transmit */
-  while(simOutSize > 0) {
+#if COOJA_SIMULATE_TURNAROUND
+  simProcessRunValue = 1;
+  cooja_mt_yield();
+  if(payload_len > 3) {
+    simProcessRunValue = 1;
     cooja_mt_yield();
   }
+#endif /* COOJA_SIMULATE_TURNAROUND */
+
+  /* Transmit on CCA */
+  if(COOJA_TRANSMIT_ON_CCA && send_on_cca && !channel_clear()) {
```

Comment:
This condition should be `if(send_on_cca && channel_clear()) {`, shouldn't it?

Reply:
I'm not following you here. The goal here is not to change the behavior of the code, just refactor it.

Do you think the proposed patch changes the behavior?
Or do you think `COOJA_TRANSMIT_ON_CCA` check shoould be removed and the `!channel_clear()` check inverted for some other reason?

Reply:
Ah, ok. Thanks. You put `COOJA_TRANSMIT_ON_CCA` in the condition instead of keeping `#if COOJA_TRANSMIT_ON_CCA` in order to make the code simpler (we don't want to return with `RADIO_TX_COLLISION` immediately).

It was a bit confusing to me that both of  `COOJA_TRANSMIT_ON_CCA` and `send_on_cca` appear in a single condition since `COOJA_TRANSMIT_ON_CCA` is used to initialize `send_on_cca`. 

> Do you think the proposed patch changes the behavior?

No.